docs/async-await.pug

```diff
@@ -0,0 +1,161 @@
+
+    ```javascript
+    async function getUser() {
+      const foundUser = await User.findOne({name: 'bill'});
+      return foundUser;
+    }
+
+    console.log(foundUser)  //Oops.  Prints '[Promise]'
+    ```
+  :markdown
+    Instead, treat the return value of an async function as you would any other promise.  Await its result inside another async function, or chain onto it using ‘.then’ blocks.
+
+    ```javascript
+    //This function returns a promise that will fulfill with the 
+    //first database entry that satisfies the search criteria.
+    async function getUser() {
+      const foundUser = await User.findOne({name: 'bill'});
+      return foundUser;
+    };
+
+    async function doStuffWithUser() {
+      //Await the promise returned from calling getUser.
+      const user = await getUser();
+
+      console.log(user);  //Prints '{name: 'bill', admin: false}'
+    }
+
+  h3(id="queries") Async/Await with Mongoose Queries
+  :markdown
+    Under the hood, [*async/await* is syntactic sugar](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await)
+    over the Promise API.  Due to the surprisingly simple way promises are implemented in JavaScript, the keyword *await* will try to unwrap any object with a property whose key is the string ‘then’ and whose value is a function.  Such objects belong to a broader class of objects often called thenables.  If the thenable being unwrapped is a genuine promise, i.e. an instance of the [Promise constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise), we enjoy several guarantees about how the object’s ‘then’ function will behave.  However, Mongoose provides several static helper methods that return a different class of thenable object called a [Query](./queries.html)--and [Queries are not promises](./queries.html#queries-are-not-promises).  Because Queries are also *thenables*, we can interact with a Query using async/await just as we would interact with a genuine promise, with one key difference: observing the fulfillment value of a genuine promise cannot under any circumstances change that value, whereas re-observing the value of a Query may cause the Query to be re-executed.
+
+    ```javascript
+    // Idiomatic way to determine if a thenable is a genuine promise
+    function isPromise(thenable) {
+        return Promise.resolve(thenable) === thenable;
+    };
+
+    // The fulfillment value of the promise returned by user.save() will always be the same,
+    // regardless of how, or how often, we observe it.
+    async function observePromise() {
+        const user = await User.findOne({firstName: 'bill', lastName:'brasky'});
+
+        user.deceased = true;
```

Comment:
Minor nitpick, but we try to avoid examples that reference sensitive topics. You could reuse the FDR example above.